virtool/analyses/data.py

```diff
@@ -69,83 +69,88 @@ def __init__(self, db: Mongo, config, pg: AsyncEngine):
-            [
-                {
-                    "$facet": {
-                        "total_count": [{"$count": "total_count"}],
-                        "found_count": [{"$count": "found_count"}],
-                        "data": [
-                            {"$sort": sort},
-                            {"$skip": skip_count},
-                            {"$limit": per_page},
-                            *lookup_nested_subtractions(),
-                            *lookup_nested_reference_by_id(),
-                        ],
-                    }
-                },
-                {
-                    "$project": {
-                        "data": {
-                            "_id": True,
-                            "workflow": True,
-                            "created_at": True,
-                            "index": True,
-                            "job": True,
-                            "ready": True,
-                            "reference": True,
-                            "sample": True,
-                            "subtractions": True,
-                            "updated_at": True,
-                            "user": True,
-                        },
-                        "total_count": {
-                            "$arrayElemAt": ["$total_count.total_count", 0]
-                        },
-                        "found_count": {
-                            "$arrayElemAt": ["$found_count.found_count", 0]
-                        },
-                    }
-                },
-            ]
-        ):
+        pipeline = [
+            {
+                "$facet": {
+                    "total_count": [{"$count": "total_count"}],
+                    "found_count": [{"$count": "found_count"}],
+                    "data": [
+                        {"$sort": sort},
+                        {"$skip": skip_count},
+                        {"$limit": per_page},
+                        *lookup_nested_subtractions(),
+                        *lookup_nested_reference_by_id(),
+                    ],
+                }
+            },
+            {
+                "$project": {
+                    "data": {
+                        "_id": True,
+                        "workflow": True,
+                        "created_at": True,
+                        "index": True,
+                        "job": True,
+                        "ready": True,
+                        "reference": True,
+                        "sample": True,
+                        "subtractions": True,
+                        "updated_at": True,
+                        "user": True,
+                    },
+                    "total_count": {"$arrayElemAt": ["$total_count.total_count", 0]},
+                    "found_count": {"$arrayElemAt": ["$found_count.found_count", 0]},
+                }
+            },
+        ]
+
+        if sample_id is not None:
+            pipeline.insert(0, {"$match": {"sample.id": sample_id}})
```

Comment:
1. I would avoid injecting a step like this, harder to debug and a nightmare if you have more than one conditional
2. I think `total_count` should probably still reflect the total number of analyses